src/MyPlot/provider/MySQLiProvider.php

```diff
@@ -77,20 +90,23 @@ public function deletePlot(Plot $plot): bool{
 	}
 
 	public function getPlot(string $levelName, int $X, int $Z): Plot{
-		if ($plot = $this->getPlotFromCache($levelName, $X, $Z)) {
+		if (($plot = $this->getPlotFromCache($levelName, $X, $Z)) instanceof Plot) {
```

Comment:
I saw several of these weird checks. As i know, if() only checks for true/false, not for null/not null